src/config_tools/config_tools.cc

```diff
@@ -364,12 +380,16 @@ void ValidateOutputs(TritonJson::Value &outs, const std::vector<IOConfig> &out_c
 
 int ReadMaxBatchSize(TritonJson::Value &config) {
   int64_t bs = -1;
-  config.MemberAsInt("max_batch_size", &bs);
+  TritonError{config.MemberAsInt("max_batch_size", &bs)};
   if (bs > std::numeric_limits<int>::max() || bs < -1) {
```

Comment:
Just confirming: -1 and 0 shouldn't really throw?

Reply:
Yeah, -1 and 0 are equivalent to missing field - in such situation we want to return -1. Less than -1 must mean  a user mistake.